### PR TITLE
[freeboxos] Fix actions to make them available in DSL rules

### LINKED ISSUE #17225: [freeboxos] Rule actions are not available in DSL rule engine
Some static methods were forgotten.

I am preparing a fix.

### PR DESCRIPTION
**Pull request by lolodomo**
Also rename actions to avoid name conflicts.

Fix #17225

Signed-off-by: Laurent Garnier <lg.hc@free.fr>

### CODE REVIEW

**Comment by lolodomo** [comment]
Looks like it can lead to a conflict of names as there is a reboot static method in both action classes:
```
Script execution of rule with UID 'freeboxos-1' failed: class org.openhab.binding.freeboxos.internal.action.ActivePlayerActions cannot be cast to class org.openhab.binding.freeboxos.internal.action.ServerActions (org.openhab.binding.freeboxos.internal.action.ActivePlayerActions and org.openhab.binding.freeboxos.internal.action.ServerActions are in unnamed module of loader org.eclipse.osgi.internal.loader.EquinoxClassLoader @13a1690) in freeboxos
```

**Comment by lolodomo** [comment]
To avoid the conflict, I have to replace in my rule:
```
val actions = getActions("freeboxos","freeboxos:active-player:api:player")
actions.reboot()
```
by
```
val actions = getActions("freeboxos","freeboxos:active-player:api:player")
org.openhab.binding.freeboxos.internal.action.ActivePlayerActions.reboot(actions)
```
I would prefer to avoid name conflicts. WDYT @clinique ? Would it be ok for you if for example we rename the methods rebootPlayer and rebootServer ?

**Review by clinique — APPROVED** [review]
LGTM

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java [hunk 1]
```diff
@@ -92,4 +92,20 @@ public void sendKeyRepeat(@ActionInput(name = "key") String key, @ActionInput(na
             logger.warn("Freebox Player Action service ThingHandler is null");
         }
     }
+
+    public static void sendKey(ThingActions actions, String key) {
+        ((PlayerActions) actions).sendKey(key);
```
Comment (jlaur) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java:97]: No need to check type?
```suggestion
        if (actions instanceof PlayerActions playerActions) {
            playerActions.sendKey(key);
        }
```

**Review by jlaur — COMMENTED** [review]
Thanks! Please see my question below.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java (hunk 1 above)
Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java:97]: In fact, this is the common pattern used in all bindings implementing actions.
I guess that the way the core framework is calling these static methods, the `actions` parameter will always be of the expected type.
Even if I agree with you that y-our proposal is better, is it a good idea to do it differently only in this binding ?

Comment (jlaur) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java:97]: Not all bindings:

https://github.com/openhab/openhab-addons/blob/858be481837ee9b45a27ffd91871728b45d1f272/bundles/org.openhab.binding.energidataservice/src/main/java/org/openhab/binding/energidataservice/internal/action/EnergiDataServiceActions.java#L257-L267

I'm pretty sure I copied this pattern from another binding, but I don't remember which. Maybe deCONZ?

https://github.com/openhab/openhab-addons/blob/858be481837ee9b45a27ffd91871728b45d1f272/bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/action/GroupActions.java#L142-L146

Comment (lolodomo) [reply bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/PlayerActions.java:97]: Ok, done.

File: bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/ActivePlayerActions.java [hunk 2]
```diff
@@ -45,7 +45,11 @@ public void reboot() {
         }
     }
 
-    public static void reboot(ThingActions actions) {
-        ((ActivePlayerActions) actions).reboot();
+    public static void rebootPlayer(ThingActions actions) {
+        if (actions instanceof ActivePlayerActions activePlayerActions) {
+            activePlayerActions.rebootPlayer();
+        } else {
+            throw new IllegalArgumentException("actions parameter is not a ActivePlayerActions class.");
```
Comment (jlaur) [inline bundles/org.openhab.binding.freeboxos/src/main/java/org/openhab/binding/freeboxos/internal/action/ActivePlayerActions.java:52]:
```suggestion
            throw new IllegalArgumentException("actions parameter is not an ActivePlayerActions class.");
```

**Review by jlaur — APPROVED** [review]
LGTM

**Pull request merged**